src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/DefaultRazorSemanticTokensInfoService.cs

```diff
@@ -555,12 +563,16 @@ internal static bool TryGetMinimalCSharpRange(RazorCodeDocument codeDocument, Ra
 
             if (csharpResponse is null)
             {
-                // C# isn't ready yet, don't make Razor wait for it
+                // C# isn't ready yet, don't make Razor wait for it. Queue up a refresh notification to ask the client
```

Comment:
Do we need to do this manually or will it just happen when the C# documents update?

Reply:
Good call, I think we can remove this case